proposals/lip-use_sha3-256_hash_of_block_header_as_blockID.md

```diff
@@ -0,0 +1,106 @@
+
+Although the success probability is very low, we want to increase the resistance against such an attack to provide sufficient security also for the future. The resistance against such an attack is determined by the bit length of the blockIDs. I.e., n-bit blockIDs yield n-bit resistance against such an attack.
+128 bit is the recommended security level by ECRYPT to provide sufficient security for at least the next [10 years
+](http://www.ecrypt.eu.org/csa/documents/D5.4-FinalAlgKeySizeProt.pdf#chapter.2). NIST is recommending 128 bit to be the minimum security level for applications intended to run [beyond 2030](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-57pt1r4.pdf#%5B%7B%22num%22%3A196%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C81%2C721%2Cnull%5D). 
+With a blockID length of 256 bit, we choose a security level that is beyond the recommendations, and provides great trust that the mentioned attack is infeasible, even for a time span that significantly exceeds the next decade.
+Choosing this extra security comes at a low price. We use 128 bit extra compared to the minimum requirement. This sums up to ~50 Mb extra size in the blockchain per year (~100 Mb if blockIDs are stored in hexadecimal representation in character arrays); a small value compared to the overall size of the chain.
+Since blockIDs are rarely handled by users, i.e. writing, reading or spelling blockIDs is a rare use case, the extra length does also not yield any significant disadvantages for the user experience.
+
+### Why SHA-3?
+
+The hash function family SHA-3, or rather the  [KECCAK](https://keccak.team/keccak.html) function, was the winner of the public NIST hash function competition. Due to the 5 year long competition and review process, it received fundamental attention and analysis. NIST chose KECCAK to be the winner of the competition due to its security, performance and flexibility aspects. We choose SHA-3 mainly for its high security due to its construction and the heavy analysis it received.
+
+#### Alternatives
+
+##### BLAKE2
+
+The hash function [BLAKE2](https://blake2.net/) received great attention and analysis as well. It is assumed to be as secure as KECCAK, and is even faster than KECCAK. However, we give preference to KECCAK as BLAKE2 did not go through the entire SHA-3 review process.
+
+##### SHA-256
+
+SHA-256 is the currently used hash function for generating blockIDs. Although there are no known weaknesses of this hash function that could be exploited with regard to blockIDs, is has weaknesses due to its construction (these can be exploited in length extension attacks). Therefore, we give a preference to KECCAK, a hash function without known weaknesses.
```

Comment:
The so called length extension _weakness_ is a result of misusing the algorithm in cases where a message authentication code should be used. I don't see how this property is relevant in our case.

Reply:
See my other answer.